### ISSUE TITLE
useless-provides even when there is no overlap/duplication

### ISSUE THREAD

**Issue by brianjmurrell** [rpm-software-management/rpmlint#895]
For a package with the following _provides_:

```
config(fuse-common) = 3.10.2-5.el8
fuse-common = 3.10.2-5.el8
fuse-common = 3.2.1
fuse-common = 3.3.0
fuse-common(x86-64) = 3.10.2-5.el8
```

`rpmlint` is complaining:

```
============================ rpmlint session starts ============================
rpmlint: 2.2.0
configuration:
    /usr/lib/python3.10/site-packages/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/licenses.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
checks: 32, packages: 1

fuse-common.x86_64: E: useless-provides fuse-common
fuse-common.x86_64: W: no-documentation
fuse-common.x86_64: W: description-shorter-than-summary
 1 packages and 0 specfiles checked; 1 errors, 2 warnings, 1 badness; has taken 0.1 s 
```

I don't see which _provides_ is actually being considered useless.

**Comment by marxin** [comment]
Please update a rpm file or a spec file I can play with.

**Comment by brianjmurrell** [comment]
https://raw.githubusercontent.com/daos-stack/fuse3/6b197b886c6df4506df6db6d4e62a7c18451cc4d/fuse3.spec

**Comment by marxin** [comment]
Can't reproduce that with the provided spec file, so please provide a RPM file.

**Comment by marxin** [comment]
You need to use any of the following formats: `ZIP (.zip, .gz)`.

**Comment by brianjmurrell** [comment]
I already tried using one of their supported extensions: .tgz:

![image](https://user-images.githubusercontent.com/559317/178573422-f38bb599-ae14-4616-96e6-84bed694a2c2.png)

and it still didn't work.

**Comment by marxin** [comment]
And what about ZIP extension?

**Comment by brianjmurrell** [comment]
Maybe.  I was really hoping to avoid having to try to iterate through re-encoding the file through the dozens of extensions that GH **purport** to support but actually reject once you actually try to use them.  If they don't actually support one of the many that they purport to support, what makes me believe that they actually support any of the others?  This just feels like an effort in vain.

.ZIP is an ugly beast fraught with proprietary extensions and really should be avoided at all costs.  Ultimately it's a pay-for-use Windows "tarball" tool that should remain the nasty world of Windows.  .tar.gz and .tgz are much more _open_ standards with which to create files.  If GH doesn't actually support them, despite saying they do, that is a real shame.

**Comment by marxin** [comment]
Fully agree that `.zip` is an ugly tarball format. Anyway, I was just able to attach a `gzip` file. All that you need to use is use `tar.gz` instead of `tgz`, see an example here:
[example-1.0-1.utm.src.rpm.gz](https://github.com/rpm-software-management/rpmlint/files/9099786/example-1.0-1.utm.src.rpm.gz)
[files.tar.gz](https://github.com/rpm-software-management/rpmlint/files/9099792/files.tar.gz)

**Comment by brianjmurrell** [comment]
Ahhh.  I guess GH need to update their list of supported extensions when an upload does fail.

[fuse-common-3.10.2-5.el8.x86_64.rpm.tar.gz](https://github.com/rpm-software-management/rpmlint/files/9102352/fuse-common-3.10.2-5.el8.x86_64.rpm.tar.gz)

**Comment by marxin** [comment]
> Ahhh. I guess GH need to update their list of supported extensions when an upload does fail.

I've just reported that to Github, it's really a bug listing `TGZ` extension.

**Comment by marxin** [comment]
About the issue, from the `useless-provides` code I see:
https://github.com/rpm-software-management/rpmlint/blob/09d9a4865b371f8abca5c38eb2f2825245e05df9/rpmlint/checks/TagsCheck.py#L669-L675

Which mentions an example of one unversioned and one versioned provides.

@Vogtinator, @DimStar77: Should we really warn about situation where one mixes versioned and not versioned provides of a same name? Is the shown example correct, I mean providing multiple versions of the same name:
```
fuse-common = 3.10.2-5.el8
fuse-common = 3.2.1
fuse-common = 3.3.0
```

?

**Comment by brianjmurrell** [comment]
But in this situation, it's not even a case of versioned and unversioned provides.  They all have a version.  The first is a version with a release and the remaining are just a version.  But they are all versioned.

**Comment by Vogtinator** [comment]
Having provides of the same name but different versions can be useful in case there is some explicit compatibility with some specific other versions, so satisfy dependencies of the `Requires: fuse-common = 3.2.1` kind. So they're not useless, but in most cases not actually necessary.
